sycl/test-e2e/format.py

```diff
@@ -232,7 +236,7 @@ def get_extra_env(sycl_devices):
                 continue
 
             # Filter commands based on split-mode
-            is_run_line = any(
+            is_run_line = unsplit_test or any(
```

Comment:
sorry so is this saying we still will report it as UNSUPPORTED by try to compile it anyway?

Reply:
on `run-only` It will no longer be marked as unsupported. Here what happens is ignore the line filtering that happens in that mode so we end up building and running the test.

Reply:
sorry, i must be missing it. where is the code that would have marked it unsupported before but not passes? the line changed here on 164 seems like it would have not been hit ever because of the `build-only` check

Reply:
So in build-only mode I needed to explicitly check for it to mark as unsupported because other features are not taken into account, for run-only mode it was originally marked as unsupported because the mode did not have the `build-and-run-mode` feature. I'm adding that feature in the lit.cfg.py so now those tests would run in run-only mode

Reply:
ah i see thx